src/item_contents.cpp

```diff
@@ -211,15 +213,14 @@ item_pocket *item_contents::best_pocket( const item &it, bool nested )
             // that needs to be something a player explicitly does
             continue;
         }
-        if( ret == nullptr ) {
-            if( pocket.can_contain( it ).success() ) {
-                ret = &pocket;
-            }
-        } else if( pocket.can_contain( it ).success() && ret->better_pocket( pocket, it ) ) {
-            ret = &pocket;
+        if( ( ret.second == nullptr && pocket.can_contain( it ).success() ) ||
+            ( pocket.can_contain( it ).success() && ret.second->better_pocket( pocket, it ) ) ) {
+            ret.first = parent;
+            ret.second = &pocket;
             for( item *contained : all_items_top( item_pocket::pocket_type::CONTAINER ) ) {
-                item_pocket *internal_pocket = contained->contents.best_pocket( it, nested );
-                if( internal_pocket != nullptr && ret->better_pocket( pocket, it ) ) {
+                std::pair<item_location, item_pocket *> internal_pocket =
+                    contained->contents.best_pocket( it, parent, nested );
```

Comment:
it looks like this is a place where you would just pass `true` for nested, because you know it is nested since you are inside the item_contents function calling this on an item.